### ISSUE TITLE
Feature Request: make public calendars writeable

### ISSUE THREAD

**Issue by tessus** [nextcloud/calendar#632]
I came across a few use cases where it would be helpful to use a shared calendar with people who

- have the link to the shared calendar
- do not have an account

What do you think? Does this make any sense? Can this be done easily or does this require a major code change?

**Comment by georgehrke** [comment]
> What do you think? Does this make any sense?

Yes, makes sense. Especially when combing with sharing only the dates but not every detail of the event. As some kind of public free busy.

> Can this be done easily or does this require a major code change?

Major as far as i can tell

**Comment by tessus** [comment]
Thanks for the reply.

> Especially when combing with sharing only the dates but not every detail of the event. As some kind of public free busy.

Although, this should be optional. In most of my use cases I want a group of people (who do not have an account) to add, change, delete events on a public calendar (maybe there's even a way to allow a combination of access permissions). The calendar is only public, if someone has the link anyway. :-)

> Major as far as i can tell

A few thoughts:
Maybe you can use some sort of virtual user in the backend, thus the logic would be the same as using a regular user. When it comes to authentication and authorization of that "public" (virtual user), it just falls through. The end user would not even have to know anything about this "user".

**Comment by n0whereRuoxi** [comment]
Did you find the solution?
We are trying to do the same job too!

**Comment by tessus** [comment]
@n0whereRuoxi I'm afraid this feature won't be happening. The `Maybe some day` milestone says it all. This is one of the issues which will be forgotten in 3 or 4 years. Priorities are often totally off with this project. (It is still not possible to add a birthday without a year to a contact.)

[RANT ON]
You're better off by moving contacts and calendar away from nextcloud. It is sad that you need several products to match rather simple requirements, but after 5 or 6 years of posting issues and trying to show off serious design flaws, I gave up. The only answers that came back were: "it's on open source project, you can contribute." Well, this is exactly the problem. I *am* contributing by helping with the design. Designers don't have to be coders. But here's the the thing: In this project all coders are designers.
[RANT OFF]

**Comment by raimund-schluessler** [comment]
> Priorities are often totally off with this project.

This issue here is number 13 on the list sorted by priority. Just because the priorities do not fit your personal preferences, I does not mean they are off.

> I am contributing by helping with the design.

Developers don't have an infinite amount of time. And although design is important and we appreciate any feedback, it is worth nothing without people having time to realize and implement (and test) it. What you are doing here, is wasting everybodys time with your rant. And obviously I am stupid enough to fall for it and answer, instead of doing something productive.

If you don't understand how free opensource programming works, you are free to use other software which fullfils your needs better (btw. that's one of the benefits of opensource, there is rarely any lockin, export your data and move on). If you find yourself paying for proprietary software in the end, then you might as well just place a bounty for this issue, and maybe someone will tackle it.

**Comment by tessus** [comment]
@raimund-schluessler I don't recall addressing my comment to you. it was clearly meant for @n0whereRuoxi 

> This issue here is number 13 on the list sorted by priority. Just because the priorities do not fit your personal preferences, I does not mean they are off.

You are wrong. Priorities in this project are seriously off. Maybe not in this case, but in a lot of other cases. This has nothing to do with my personal preferences, but with the fact that in certain cases they affect me as well.

A few examples:

- Even when fixes are available, they are ignored: https://github.com/nextcloud/server/issues/5653
- The group admin disaster (the first time I opened this was 4 years ago in the owncloud repo): https://github.com/nextcloud/server/issues/3630 (this issue has been dragged along and discussed for years)
- even when it's marked as a bug and there's real data loss involved - nothing: https://github.com/nextcloud/server/issues/4287

Shall I go on?

However, if there are cosmetic GUI issues, which aren't real problems, they are always fixed first. Don't tell me because it's easier to tackle low hanging fruits. In this case more complex issues are never going to be fixed (well, as you can see from my list above, I'm right there too).

> If you don't understand how free opensource programming works

I do. And I fix current issues first, before I start developing new half baked solutions. I also make sure that basic requirements are met, before I call it production ready.

@n0whereRuoxi btw, I forgot to mention. Do not share your opinion with any of the developers and don't talk about facts. A lot of developers see that as a personal attack and think you are doing this to hurt their feelings.

**Comment by marbetschar** [comment]
**TL;DR:** Public share functionality seems to be broken.

**All details**
Tried to create a public calendar which I wanted to share with a friend w/o Nextcloud user. Didn't work at all. Here's what I did:

* Create a new calendar
* Generate a public share link for the newly created calendar
* Sent the public share link to my friend

When my friend openend the public share link on his iOS device - where Safari offers to "Download" the calendar events and importing them into an existing calendar on the device. That happens for both options, regardless whether he clicks "Download" or "Copy Subscription Link" in the hamburger menu of the public calendar.

What I expected is the possibility to "Subscribe" to the calendar, not to just import the events once. Along with this, I also expected a way to allow him to edit the calendar - which obviously does not work too.

**Environment information**
- Using Nextcloud 17.0.2.1
- Created and shared calendar in web UI, using Firefox v73.0.1
- Opened public calendar in Safari on iOS 13.3.1

**Comment by georgehrke** [comment]
@marbetschar Please don't hijack other issues. If you found a bug, open a new issue. Thx

**Comment by marbetschar** [comment]
@georgehrke tried to create a new calendar, share it in public and allow an anonymous user to edit it. Didn't work. Thought that's close enough related to this issue.

However, I'm happy to open another calendar sharing with edit permission issue if that's of any help.

**Comment by georgehrke** [comment]
> TL;DR: Public share functionality seems to be broken.

You said public sharing is broken. In case public sharing links don't work for you at all, yes, please open a new issue.

If this is just about the missing ability to edit:
That's not broken but simply something that has not been implemented so far. Hence this Feature request is still open.

**Comment by georgehrke** [comment]
Just noticed that this is a duplicate of https://github.com/nextcloud/server/issues/3509
